app/assets/javascripts/admin/order_cycles/controllers/edit.js.coffee

```diff
@@ -18,5 +19,14 @@ angular.module('admin.orderCycles')
 
     $scope.submit = ($event, destination) ->
       $event.preventDefault()
+      $scope.navClear()
       StatusMessage.display 'progress', t('js.saving')
       OrderCycle.update(destination, $scope.order_cycle_form)
+
+    # Provide a callback for generating warning messages displayed before leaving the page. 
+    orderCycleNavCallback = ->
+      if $scope.order_cycle_form != undefined && $scope.order_cycle_form.$dirty
```

Comment:
coffee script has the safe navigation operator like ruby but it's with a question mark.
this becomes
f $scope.order_cycle_form?.$dirty

Reply:
can be changed on enterpriseCtrl as well

Reply:
Ok but like `.ts` - is that effectively saying, I'm certain that this exists? If so are we certain of that?

Reply:
Its saying "I am not certain that this property exists, so return false if it doesnt exist or navigate further (to $dirty) if it does"

Reply:
Ace

Reply:
Fixed with 7a41494